### PR TITLE
Add annotated tagging to deployments

### PR DESCRIPTION
**Pull request by chiting**
This PR adds annotated tagging when deploying projects to environments.

Notes:
* Added an additional field to the configuration, `RH_USER_EMAIL_DOMAIN`

**90%**

Note bis: production workflow will be implemented as soon as we are happy with how it works on QA and UAT boxes

---
#### TODO:
- [x] ~Add production flow~
- [x] Sentry integration (https://github.com/globaldev/releasehub/pull/13)

### CODE REVIEW

File: app/helpers/deployments_helper.rb [hunk 1]
```diff
@@ -1,4 +1,94 @@
...
+          create_annotated_tag_reference(
+            tag_name, annotated_tag[:sha], repo_name
+          )
+        end
+      end
+    end
+  end
+
+  def detach_deployment_tags(deployment)
+    if deployment.environment.production?
```
Comment (chiting) [inline app/helpers/deployments_helper.rb]: The production logic will be implemented later, once we've battle tested this feature on QA/UAT.

File: app/helpers/deployments_helper.rb [hunk 2]
```diff
@@ -1,4 +1,94 @@
 module DeploymentsHelper
   DEFAULT_NOTIFY_IDS = ENV["DEFAULT_NOTIFY_IDS"]
   DEFAULT_CHANNEL = ENV["DEFAULT_CHANNEL"]
+  USER_EMAIL_DOMAIN = ENV["RH_USER_EMAIL_DOMAIN"]
+
+  def attach_deployment_tags(deployment, ops)
```
Comment (mendable) [inline app/helpers/deployments_helper.rb]: These are application level concerns, not view related. It's bad that view helpers are already being included within `ApplicationController` - in most/all cases they probably shouldn't be.

I'd suggest moving all of this logic to classes within app/models/.

Comment (chiting) [inline app/helpers/deployments_helper.rb]: Only the tagging ones or all of them?

**Comment by chiting** [comment]
Cleaning up my PR lists a bit (and this one looks stale :)

**Pull request closed without merging**